x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/pipelines_json_configurations.tsx

```diff
@@ -0,0 +1,142 @@
+            {i18n.translate(
+              'xpack.enterpriseSearch.content.indices.pipelines.tabs.jsonConfigurations.ingestionPipelines.docLink',
+              {
+                defaultMessage: 'Learn more about how Enterprise Search uses ingest pipelines',
+              }
+            )}
+          </EuiLink>
+        }
+        action={
+          pipelineNames.length > 0 && (
+            <EuiNotificationBadge size="m">{pipelineNames.length}</EuiNotificationBadge>
+          )
+        }
+        iconType="visVega"
+      >
+        <EuiFormRow
+          fullWidth
+          label={i18n.translate(
+            'xpack.enterpriseSearch.content.indices.pipelines.tabs.jsonConfigurations.selectLabel',
+            { defaultMessage: 'Select an ingest pipeline to view' }
+          )}
+        >
+          <EuiSelect
+            fullWidth
+            options={pipelineNames.map((name) => ({ text: name, value: name }))}
+            value={selectedPipelineId}
+            onChange={(e) => selectPipeline(e.target.value)}
+          />
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        {selectedPipeline && (
+          <>
+            <EuiFlexGroup alignItems="center">
+              <EuiFlexItem>
+                <PipelineJSONBadges />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                {isManagedPipeline(selectedPipeline) ? (
+                  <EuiButtonEmpty
+                    size="s"
+                    flush="both"
+                    iconType="eye"
+                    color="primary"
+                    href={http.basePath.prepend(
+                      `/app/management/ingest/ingest_pipelines/?pipeline=${selectedPipelineId}`
+                    )}
+                  >
+                    {i18n.translate(
+                      'xpack.enterpriseSearch.content.indices.pipelines.tabs.jsonConfigurations.action.view',
+                      {
+                        defaultMessage: 'View in Stack Management',
+                      }
+                    )}
+                  </EuiButtonEmpty>
+                ) : (
+                  <EuiButtonEmpty
+                    size="s"
+                    flush="both"
+                    iconType="pencil"
+                    color="primary"
+                    href={http.basePath.prepend(
+                      `/app/management/ingest/ingest_pipelines/edit/${selectedPipelineId}`
+                    )}
+                  >
+                    {i18n.translate(
+                      'xpack.enterpriseSearch.content.indices.pipelines.tabs.jsonConfigurations.action.view',
+                      {
+                        defaultMessage: 'Edit in Stack Management',
+                      }
+                    )}
+                  </EuiButtonEmpty>
+                )}
+              </EuiFlexItem>
+            </EuiFlexGroup>
+            <EuiSpacer size="m" />
+            <EuiCodeBlock language="json" overflowHeight={300} isCopyable>
```

Comment:
this `overflowHeight={300}` is a bit arbitrary, is there a patter we usually follow for what to set this value to?

Reply:
I don't think we have, we can try to come up with one though.